cmd/vsphere-xcopy-volume-populator/internal/powerstore/powerstore.go

```diff
@@ -0,0 +1,329 @@
+package powerstore
+
+import (
+	"context"
+	"fmt"
+	"slices"
+	"strings"
+
+	"github.com/dell/gopowerstore"
+	"github.com/kubev2v/forklift/cmd/vsphere-xcopy-volume-populator/internal/populator"
+	"k8s.io/klog/v2"
+)
+
+const (
+	initiatorGroupContextKey string = "initiatorGroup"
+	hostIDContextKey         string = "hostID"
+	hostNameContextKey       string = "hostName"
+)
+
+type PowerstoreClonner struct {
+	Client gopowerstore.Client
+}
+
+// CurrentMappedGroups implements populator.StorageApi.
+func (p *PowerstoreClonner) CurrentMappedGroups(targetLUN populator.LUN, mappingContext populator.MappingContext) ([]string, error) {
+	if targetLUN.IQN == "" {
+		return nil, fmt.Errorf("target LUN IQN is required")
+	}
+
+	ctx := context.Background()
+	mappings, err := p.Client.GetHostVolumeMappingByVolumeID(ctx, targetLUN.IQN)
+	if err != nil {
+		return nil, fmt.Errorf("failed to get volume mappings for %s: %w", targetLUN.Name, err)
+	}
+
+	mappedHosts := make([]string, 0, len(mappings))
+	for _, mapping := range mappings {
+		host, err := p.Client.GetHost(ctx, mapping.HostID)
+		if err != nil {
+			klog.Warningf("Failed to get host info for host ID %s: %s", mapping.HostID, err)
+			continue
+		}
+		mappedHosts = append(mappedHosts, host.Name)
+	}
+
+	klog.Infof("Volume %s is currently mapped to hosts: %v", targetLUN.Name, mappedHosts)
+	hostName, ok := mappingContext[hostNameContextKey].(string)
+	if !ok || hostName == "" {
+		return nil, fmt.Errorf("mappingContext missing %q", hostNameContextKey)
```

Comment:
nit: its missing either the key, or the value is "" , hence no hostname set there.